api/v1alpha4/condition_consts.go

```diff
@@ -184,3 +184,38 @@ const (
 	// from making any further remediations.
 	TooManyUnhealthyReason = "TooManyUnhealthy"
 )
+
+// These are valid conditions of a MachineDeployment.
+const (
+	// Available means the MachineDeployment is available, ie. at least the minimum available
+	// machines required are up and running for at least minReadySeconds.
+	MachineDeploymentAvailable ConditionType = "MachineDeploymentAvailable"
+	// Progressing means the MachineDeployment is progressing. Progress for a MachineDeployment is
+	// considered when a new machine set is created or adopted, and when new machine scale
+	// up or old machine scale down. Progress is not estimated for paused deployments or
+	// when progressDeadlineSeconds is not specified.
+	MachineDeploymentProgressing ConditionType = "MachineDeploymentProgressing"
+	// MachineFailure is added in a MachineDeployment when one of its machines fails to be created
+	// or deleted.
+	MachineDeploymentMachineFailure ConditionType = "MachineDeploymentMachineFailure"
+
+	// Reasons for deployment conditions
+	//
+	// Progressing:
+	// NewMSAvailableReason is added in a deployment when its newest machine set is made available
+	NewMSAvailableReason = "NewMachineSetAvailable"
+	// TimedOutReason is added in a deployment when its newest machine set fails to show any progress
+	// within the given deadline (progressDeadlineSeconds).
+	TimedOutReason = "ProgressDeadlineExceeded"
+	// MachineSetUpdatedReason is added in a deployment when one of its machine sets is updated as part
+	// of the rollout process.
+	MachineSetUpdatedReason = "MachineSetUpdated"
+	//
+	// Available:
+
+	// MinimumMachinesAvailable is added in a deployment when it has its minimum machines required available.
+	MinimumMachinesAvailable = "MinimumMachinesAvailable"
+	// MinimumMachinesUnavailable is added in a deployment when it doesn't have the minimum required machines
+	// available.
+	MinimumMachinesUnavailable = "MinimumMachinesUnavailable"
```

Comment:
nit: Is it possible to map reasons and conditions, so it will be easier to understand the expected workflow/state diagram for each condition?